client.go

```diff
@@ -205,151 +201,97 @@ func (r *marathonClient) Ping() (bool, error) {
-			}
+	var jsonBody []byte
+	if body != nil {
+		jsonBody, err = json.Marshal(body)
+		if err != nil {
+			return err
 		}
-		return status, content, nil
 	}
 
-	switch status {
-	case 500:
-		return status, "", ErrInvalidResponse
-	case 404:
-		return status, "", ErrDoesNotExist
-	case 409:
-		return status, "", ErrConflict
+	// Make the http request to Marathon
+	request, err := http.NewRequest(method, url, bytes.NewReader(jsonBody))
+	if err != nil {
+		return err
 	}
 
-	// step: lets decode into a error message
-	var message struct {
-		Message string `json:"message"`
+	// Add any basic auth and the content headers
+	if r.config.HTTPBasicAuthUser != "" {
+		request.SetBasicAuth(r.config.HTTPBasicAuthUser, r.config.HTTPBasicPassword)
 	}
+	request.Header.Add("Content-Type", "application/json")
+	request.Header.Add("Accept", "application/json")
 
-	if err := r.decodeRequest(strings.NewReader(content), &message); err != nil {
-		return status, content, ErrInvalidResponse
+	response, err := r.httpClient.Do(request)
+	if err != nil {
+		return err
 	}
+	defer response.Body.Close()
 
-	errorMessage := "unknown error"
-	if message.Message != "" {
-		errorMessage = message.Message
+	respBody, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return err
 	}
 
-	return status, "", fmt.Errorf("%s", errorMessage)
-}
-
-func (r *marathonClient) httpRequest(method, uri, body string) (int, string, *http.Response, error) {
-	var content string
-	var response *http.Response
-
-	// Try to connect to Marathon until succeed or
-	// the whole custer is down
-	for {
-		// Get a member from the cluster
-		marathon, err := r.cluster.GetMember()
-		if err != nil {
-			return 0, "", nil, err
-		}
-
-		url := fmt.Sprintf("%s/%s", marathon, uri)
+	if len(jsonBody) > 0 {
+		log.Printf("apiCall(): %v %v %s returned %v %s\n", request.Method, request.URL.String(), jsonBody, response.Status, oneLogLine(respBody))
+	} else {
+		log.Printf("apiCall(): %v %v returned %v %s\n", request.Method, request.URL.String(), response.Status, oneLogLine(respBody))
+	}
 
-		glog.V(debugLevel).Infof("[http] request: %s, uri: %s, url: %s", method, uri, url)
-		// Make the http request to Marathon
-		request, err := http.NewRequest(method, url, strings.NewReader(body))
-		if err != nil {
-			return 0, "", nil, err
+	if response.StatusCode >= 200 && response.StatusCode <= 299 {
+		if result != nil {
+			if err := json.Unmarshal(respBody, result); err != nil {
+				log.Printf("apiCall(): failed to unmarshall the response from marathon, error: %s\n", err)
+				return ErrInvalidResponse
+			}
 		}
+		return nil
 
-		// Add any basic auth and the content headers
-		if r.config.HTTPBasicAuthUser != "" {
-			request.SetBasicAuth(r.config.HTTPBasicAuthUser, r.config.HTTPBasicPassword)
-		}
-		request.Header.Add("Content-Type", "application/json")
-		request.Header.Add("Accept", "application/json")
+	} else if response.StatusCode == 404 {
+		return ErrDoesNotExist
```

Comment:
The switch-case syntax was more concise and thus more readable. Is there any particular reason you had to replace it?